### PR TITLE
Development

### PR DESCRIPTION
**Pull request by singhadi672**
please review my code to be able to merge into main

### CODE REVIEW

File: Models/user.model.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  history: [{ type: mongoose.Schema.Types.ObjectId, ref: "Video" }],
+});
+
+const User = mongoose.model("User", UserSchema);
+
+module.exports = { User };
+
+
+
+// async function dasa() {
```
Comment (duhbhavesh) [inline Models/user.model.js:22]: comments can be removed

File: Routes/history.router.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            .status(201)
+            .json({ status: false, message: "video already present" });
+        } else {
+          history = [...history, newVideo.id];
+          const response = await User.updateOne({ history: history });
+          res.status(201).json({ status: true, response });
+        }
+      }
+    } catch (err) {
+      res.status(500).json({ success: false, errmessage: err });
```
Comment (duhbhavesh) [inline Routes/history.router.js]: `errmessage` can be written as `errMessage` for readability.

```suggestion
      res.status(500).json({ success: false, errMessage: err });
```

File: Routes/likedVideos.router.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        const filteredArray = likedVideos.filter((item) => item != newVideo.id);
+        const response = await User.updateOne({ likedVideos: filteredArray });
+        res.status(201).json({ status: true, response });
+      } else {
+        likedVideos = [...likedVideos, newVideo.id];
+        const response = await User.updateOne({ likedVideos: likedVideos });
+        res.status(201).json({ status: true, response });
+      }
+    } catch (err) {
+      res.status(500).json({ success: false, errmessage: err });
```
Comment (duhbhavesh) [inline Routes/likedVideos.router.js:40]: Here as well.

**Review by duhbhavesh — COMMENTED** [review]
Hi @singhadi672 

The code looks very well written, I've just suggested few minor changes.

File: DB/DBConnection.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require("mongoose");
+const dotenv = require("dotenv");
+dotenv.config();
+const DB_USERNAME = process.env.mongoDBUsername;
+const DB_PASSWORD = process.env.mongoDBPassword;
+const dbName = "video-library";
```
Comment (barbaricankit) [inline DB/DBConnection.js:2]: nice

File: Models/user.model.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+//   const data = {
+//     likedVideos: [{ _id }],
+//     playlist: [{ playlistName: "My Playlist", playlistVideos: [{ _id }] }],
+//     watchLater: [{ _id }],
+//     history: [{ _id }],
+//   };
+
+//   const user = new User(data);
+//   const resp = await user.save();
+//   console.log(resp);
```
Comment (barbaricankit) [inline Models/user.model.js:33]: remove comments

File: Routes/history.router.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  .route("/")
+  .get(async (req, res) => {
+    const { history } = await User.findOne().populate("history").exec();
+    res.status(200).json({ status: true, history });
+  })
+  .post(async (req, res) => {
+    try {
+      const newVideo = req.body;
+      if (newVideo.clear) {
+        const response = await User.updateOne({ history: [] });
```
Comment (barbaricankit) [inline Routes/history.router.js]: nice

File: Routes/playlist.router.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        const response = await User.updateOne({ playlist: filteredPlaylist });
+        res.status(201).json({ status: true, response });
+      } else {
+        playlist = [
+          ...playlist,
+          { playlistName: playlistReq.playlistName, playlistVideos: [] },
+        ];
+        const response = await User.updateOne({ playlist: playlist });
+        res.status(201).json({ status: true, response });
+      }
```
Comment (barbaricankit) [inline Routes/playlist.router.js]: use middleware

File: Routes/playlist.router.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    const { playlist } = await User.findOne();
+    let targetPlaylist = playlist.find(
+      (item) => item._id == videoReq.playlistId
+    );
+
+    const isVideoPresent = targetPlaylist.playlistVideos.find(
+      (item) => item == videoReq.videoId
+    );
+
+    console.log(isVideoPresent);
```
Comment (barbaricankit) [inline Routes/playlist.router.js]: remove console

File: Routes/playlist.router.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+    if (isVideoPresent) {
+      const filteredArray = targetPlaylist.playlistVideos.filter(
+        (item) => item != videoReq.videoId
+      );
+      console.log(filteredArray);
```
Comment (barbaricankit) [inline Routes/playlist.router.js:76]: same here

**Review by barbaricankit — COMMENTED** [review]
Nice job man.. Just one suggestion try to use more middlewares

**Pull request merged**